scripts/script.js

```diff
@@ -1,118 +1,92 @@
 $(function () {
 
-    var carouselList = $('#js-myCarousel ul'),
-        interval = setInterval(changeSlideLeft, 1000),
-        i = 0,
-        $activeSlide = $('.indicators .item');
+    var carouselList = $('#js-myCarousel ul');
+    var currentSlideIndex = 0;
+    var $indicators = $('.indicators .item');
+    var lastSlideIndex = $indicators.length - 1;
+    // var interval = setInterval(function() {
+    //     moveSlidesForward(1);
+    // }, 1000);
 
 //start karuzeli//
 
-    function changeSlideLeft() {
-        if (i === 4) {
-            clearInterval(interval);
-            interval = setInterval(changeSlideRight, 1000);
+    function moveSlidesForward(numberOfSlides) {
+        if (currentSlideIndex === lastSlideIndex) {
             return;
         }
-        carouselList.animate({'margin-left': -400}, 500, moveFirstSlide);
-        i++;
-        setActiveSlide(i);
+        carouselList.animate({'margin-left': -400 * numberOfSlides}, 500, function() {
+            moveSlidesFromStart(numberOfSlides);
```

Comment:
`numberOfSlides` bardzo często pojawia się w kodzie. Rozumiem że to nie ma wartości wyjściowej a jedynie nadajemy ją tam gdzie będzie nam potrzebna?